src/pip/_internal/operations/check.py

```diff
@@ -71,6 +73,9 @@ def check_package_set(
         for req in package_detail.dependencies:
             name = canonicalize_name(req.name)
 
+            if should_ignore_dependencies and should_ignore_dependencies(name):
+                continue
```

Comment:
Should we log this? e.g. `log.debug("%s was ignored because --ignore-packages is set", name)` or something similar?

Reply:
Added